core/src/engine/EngineMultiRobot.cc

```diff
@@ -941,6 +950,15 @@ namespace jiminy
             tEnd = stepperState_.t + stepSize_true;
             stepperState_.tError = (tEnd - stepperState_.t) - stepSize_true;
 
+            // Check that tEnd is not too large for the current logging precision.
+            if (tEnd > telemetryRecorder_->getMaximumLogTime())
```

Comment:
line 951 `stepperState_.tError = (tEnd - stepperState_.t) - stepSize_true;`  must be moved after this block.